File: 014. Longest Common Prefix/Java/Solution.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0) {
            return "";
        }
        String firstStr = strs[0];
        int commonIndex = 0;
        while (true) {
            for (String str : strs) {
                if (str.length() == commonIndex || str.charAt(commonIndex) != firstStr.charAt(commonIndex)) {
                    return firstStr.substring(0, commonIndex);
                }
            }
            commonIndex++;
        }
    }
}

File: 014. Longest Common Prefix/C++/Solution.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string> &strs) {
        if (strs.empty()) {
            return "";
        }
        string firstStr = strs[0];
        int commonIndex = 0;
        while (true) {
            for (string &str : strs) {
                if (str.size() == commonIndex || str[commonIndex] != firstStr[commonIndex]) {
                    return firstStr.substr(0, commonIndex);
                }
            }
            commonIndex++;
        }
    }
};